lib/entice/logic/vitals.ex

```diff
@@ -4,61 +4,139 @@ defmodule Entice.Logic.Vitals do
+  """
+  def heal(entity, amount),
+  do: Coordination.notify({:vitals_entity_heal, ammount}, entity)
 
-  defmodule Behaviour do
+  @doc """
+  Resurrects an entity with a Percentage of Life Points and Energy
+  New Health and Energy is dependent of Morale
+  """
+  def resurrect(entity, percent_health, percent_energy),
+  do: Coordination.notify({:vitals_entity_resurrect, percent_health, percent_energy}, entity)
+
+  defmodule AliveBehaviour do
     use Entice.Entity.Behaviour
     alias Entice.Logic.Vitals.Health
     alias Entice.Logic.Vitals.Energy
+    alias Entice.Logic.Vitals.Morale
     alias Entice.Logic.Player.Level
 
+    def init(%Entity{attributes: %{Level => %Level{level: _}}} = entity, {:entity_resurrected, percent_health, percent_energy}) do
+      {_, max_health: max_health} = get_max_health(entity)
+      resurrected_health = max_health / 100 * percent_health
 
-    def init(entity, _args) do
-      {:ok, entity |> put_attribute(get_max_health(entity))
-                   |> put_attribute(get_max_energy(entity))}
+      {_, max_mana: max_mana} = get_max_energy(entity)
+      resurrected_mana = max_mana / 100 * percent_energy
 
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: resurrected_health, max_health: max_health} end)
+                   |> update_attribute(Energy, fn _ -> %Energy{mana: resurrected_mana, max_mana: max_mana} end)}
     end
 
+    def init(%Entity{attributes: %{Level => %Level{level: _}}} = entity, _args) do
+      {:ok, entity |> put_attribute(get_max_health(entity))
+                   |> put_attribute(get_max_energy(entity))
+                   |> put_attribute(%Morale{morale: 0})}
+    end
+
+    def terminate(:remove_handler, entity) do
+      {:ok, entity}
+    end
 
     def terminate(_reason, entity) do
       {:ok, entity |> remove_attribute(Health)
-                   |> remove_attribute(Energy)}
+                   |> remove_attribute(Energy)
+                   |> remove_attribute(Morale)}
+    end
+
+    def handle_event({:vitals_entity_damage, amount}, %Entity{attributes: %{Health => %Health{health: health, max_health: _}}} = entity) do
+      new_health = health - amount
+      cond do
+        new_health <= 0 -> {:become, Vitals.DeadBehaviour, {entity, :vitals_entity_died}, entity |> update_attribute(Health, fn health -> %Health{health: 0, max_health: health.max_health} end)}
+        new_health > 0 -> {:ok, entity |> update_attribute(Health, fn health -> %Health{health: new_health, max_health: health.max_health} end)}
+      end
     end
 
+    def handle_event({:vitals_entity_heal, amount}, %Entity{} = entity) do
+      {:ok, %Health{health: health, max_health: max_health}} = fetch_attribute(entity, Health)
+      new_health = health + amount
+      if new_health > max_health do
+        new_health = max_health
+      end
 
-    # internal
+      {:ok, entity |> update_attribute(Health, fn _ -> %Health{health: new_health, max_health: max_health} end)}
```

Comment:
use the passed in %Health{... here:
```Elixir
fn health -> %Health{health | health: new_health} end
``` 
(this is how you can update maps idiomatically in elixir)